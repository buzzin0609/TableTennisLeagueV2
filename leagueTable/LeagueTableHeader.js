/**
 * Description: Header for the league table
 */

import React, { Component } from 'react';
import HeadingType from "./HeadingType";
import Heading from "./Heading";
import {Grid} from "react-native-easy-grid";
import {Container, ListItem} from "native-base";

type Props = {
	headings: HeadingType[]
};

class LeagueTableHeader extends Component<Props> {

  render() {
    return (
        <Grid>
			{
				this.props.headings.map((heading: HeadingType): Heading => {
					return <Heading key={heading.slug} {...heading} />;
				})
			}

		</Grid>
    )
  }
}

export default LeagueTableHeader