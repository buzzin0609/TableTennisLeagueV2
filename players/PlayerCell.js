/**
 * Description: Table cell for player info
 */

import React, { Component } from 'react';
import { View } from 'react-native';
import type {PlayerModel} from "./Player";
import type {HeadingModel} from "../leagueTable/HeadingType";
import {Col, Grid, Row} from "react-native-easy-grid";
import playerCellStyles from "./playerCellStyles";
import HeadingType from "../leagueTable/HeadingType";
import Text from "../layout/Text";
import {Container, ListItem} from "native-base";

type Props = {
	player: PlayerModel,
	columns: HeadingType[]
}

class PlayerCell extends Component<Props> {
  state = {};

  render() {
    return (
		<Grid>
			{
				this.props.columns.map((heading: HeadingType): Col => {
					return (
						<Col key={heading.slug} style={playerCellStyles.cell} size={heading.size}>
							<Text style={playerCellStyles.text}>{this.props.player[heading.slug]}</Text>
						</Col>
					);
				})
			}
		</Grid>
    )
  }
}

export default PlayerCell
