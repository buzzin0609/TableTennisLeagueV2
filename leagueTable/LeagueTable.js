/**
 * Description: Main league table
 */

import React, { Component } from 'react';
import {Container, Content, List} from "native-base";
import LeagueTableHeader from "./LeagueTableHeader";
import headingList from './headingList';
import type {PlayerModel} from "../players/Player";
import LeagueTableBody from "./LeagueTableBody";
import {Grid} from "react-native-easy-grid";
import PlayerType from "../players/Player";
import StateWrapper from "../state/StateWrapper";

class LeagueTable extends Component<Props> {
  render() {
    return [
		<LeagueTableHeader key={'league-header'} headings={headingList} />,
		<StateWrapper key={`league-body`} stateProps={'players'}>
			<LeagueTableBody />
		</StateWrapper>
	]
  }
}

export default LeagueTable