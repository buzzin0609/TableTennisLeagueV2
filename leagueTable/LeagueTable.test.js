import React from 'react';
import LeagueTable from './LeagueTable';

import renderer from 'react-test-renderer';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import LeagueTableHeader from "./LeagueTableHeader";
import PlayerCell from "../players/PlayerCell";
import type {PlayerModel} from "../players/Player";
import PlayerType from "../players/Player";

Enzyme.configure({adapter: new Adapter()});

it('renders without crashing', () => {
	const rendered = renderer.create(<LeagueTable/>).toJSON();
	expect(rendered).toMatchSnapshot();
});

it('should render player cells when players passed to LeagueTable', function () {
	const model: PlayerModel = {
		name: 'Foo',
		wins: 5,
		losses: 0,
		rankPoints: 0,
		gamesPlayed: 5
	};
	const player: PlayerType = new PlayerType(model)
	const view = shallow(<LeagueTable />);

	expect(view).toMatchSnapshot();

	view.setProps({
		players: [player]
	});

	expect(view).toMatchSnapshot();
});