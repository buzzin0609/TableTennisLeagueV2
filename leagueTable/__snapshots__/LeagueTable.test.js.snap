// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    size={2}
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "flex": 2,
        "flexDirection": "row",
        "padding": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "roboto-regular",
          "fontSize": 16,
        }
      }
    >
      Player
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
  <View
    size={1}
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "padding": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "roboto-regular",
          "fontSize": 16,
        }
      }
    >
      Pld
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
  <View
    size={1}
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "padding": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "roboto-regular",
          "fontSize": 16,
        }
      }
    >
      W
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
  <View
    size={1}
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "padding": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "roboto-regular",
          "fontSize": 16,
        }
      }
    >
      L
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
  <View
    size={1.5}
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "flex": 1.5,
        "flexDirection": "row",
        "padding": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "roboto-regular",
          "fontSize": 16,
        }
      }
    >
      RP
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
</View>
`;

exports[`should render player cells when players passed to LeagueTable 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LeagueTable />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <LeagueTableHeader
      headings={
        Array [
          HeadingType {
            "name": "Player",
            "size": 2,
            "slug": "name",
          },
          HeadingType {
            "name": "Pld",
            "size": 1,
            "slug": "gamesPlayed",
          },
          HeadingType {
            "name": "W",
            "size": 1,
            "slug": "wins",
          },
          HeadingType {
            "name": "L",
            "size": 1,
            "slug": "losses",
          },
          HeadingType {
            "name": "RP",
            "size": 1.5,
            "slug": "rankPoints",
          },
        ]
      }
    />,
    <StateWrapper
      stateProps="players"
    >
      <LeagueTableBody
        players={Array []}
      />
    </StateWrapper>,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <LeagueTableHeader
        headings={
          Array [
            HeadingType {
              "name": "Player",
              "size": 2,
              "slug": "name",
            },
            HeadingType {
              "name": "Pld",
              "size": 1,
              "slug": "gamesPlayed",
            },
            HeadingType {
              "name": "W",
              "size": 1,
              "slug": "wins",
            },
            HeadingType {
              "name": "L",
              "size": 1,
              "slug": "losses",
            },
            HeadingType {
              "name": "RP",
              "size": 1.5,
              "slug": "rankPoints",
            },
          ]
        }
      />,
      <StateWrapper
        stateProps="players"
      >
        <LeagueTableBody
          players={Array []}
        />
      </StateWrapper>,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should render player cells when players passed to LeagueTable 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LeagueTable
    players={
      Array [
        Player {
          "gamesPlayed": 5,
          "losses": 0,
          "name": "Foo",
          "rankPoints": 0,
          "rankPointsPrevious": 0,
          "wins": 5,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <LeagueTableHeader
      headings={
        Array [
          HeadingType {
            "name": "Player",
            "size": 2,
            "slug": "name",
          },
          HeadingType {
            "name": "Pld",
            "size": 1,
            "slug": "gamesPlayed",
          },
          HeadingType {
            "name": "W",
            "size": 1,
            "slug": "wins",
          },
          HeadingType {
            "name": "L",
            "size": 1,
            "slug": "losses",
          },
          HeadingType {
            "name": "RP",
            "size": 1.5,
            "slug": "rankPoints",
          },
        ]
      }
    />,
    <StateWrapper
      stateProps="players"
    >
      <LeagueTableBody
        players={Array []}
      />
    </StateWrapper>,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <LeagueTableHeader
        headings={
          Array [
            HeadingType {
              "name": "Player",
              "size": 2,
              "slug": "name",
            },
            HeadingType {
              "name": "Pld",
              "size": 1,
              "slug": "gamesPlayed",
            },
            HeadingType {
              "name": "W",
              "size": 1,
              "slug": "wins",
            },
            HeadingType {
              "name": "L",
              "size": 1,
              "slug": "losses",
            },
            HeadingType {
              "name": "RP",
              "size": 1.5,
              "slug": "rankPoints",
            },
          ]
        }
      />,
      <StateWrapper
        stateProps="players"
      >
        <LeagueTableBody
          players={Array []}
        />
      </StateWrapper>,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
